app.js

```diff
@@ -9,11 +8,71 @@ class Dashboard extends React.Component {
     /**
      * @inheritdoc
      */
+    constructor(props) {
+        super(props);
+        this.psnrTestResultTransformFunc = jsonData => {
+            const extractData = (json, yAxisFieldName) =>
+                json.reduce((currData, currRow) => {
+                    const field = currRow[yAxisFieldName];
+
+                    currData.push([ currRow.buildNum, field ]);
+
+                    return currData;
+                }, []);
+
+            // The frame data needs to be transformed to the percentage
+            //  of all the frames, so we need another helper
+            const extractPctData
+                = (json,
+                        yAxisNumeratorFieldName,
+                        yAxisDenominatorFieldName) =>
+                    json.reduce((currData, currRow) => {
+                        const num = currRow[yAxisNumeratorFieldName];
+                        const denom = currRow[yAxisDenominatorFieldName];
+                        const val = num / denom;
+
+                        currData.push([ currRow.buildNum, val ]);
+
+                        return currData;
+                    }, []);
+
+            const psnrData = extractData(jsonData, 'psnr');
+            const frameSkipData
+                = extractPctData(jsonData, 'numSkippedFrames', 'totalFrames');
+            const frameFrozenData
+                = extractPctData(jsonData, 'numFrozenFrames', 'totalFrames');
```

Comment:
This almost sounds like it needs a chart model to abstract away this info.

Reply:
you mean a chart model like...the ones i had that you didn't like? :)

but yeah, all this here doesn't feel good and feels like it should be somewhere more specific

Reply:
To be fair, I think previously you were mixing view logic with model logic.